pkg/client/requestor.go

```diff
@@ -0,0 +1,36 @@
+/*
+Copyright (c) 2018 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package client contains the types and functions used to communicate with other services using
+// queues and topics.
+package client
+
+// ResponseCallback is called when a response to a request is received
+// m is the response message
+// requestID is the id of the request this message responds
+type ResponseCallback func(response Message, requestID string) error
+
+// RequestorSpec is a helper struct for building requestors.
+type RequestorSpec struct {
+	Destination   string
+	ResponseQueue string
+}
+
+// Requestor is a specification of publish/subscribe mechanism
+// It allows sending direct response and supply a callback
+type Requestor interface {
+	Send(request Message, callback ResponseCallback) (requestID string, err error)
```

Comment:
Why is the `requestID` needed here? Can we remove it?

Reply:
I think we talked about removing it in the meeting, ideally the library should generate request IDs itself.

Reply:
As we discussed in the meeting the request ID is generated by the library, and returned as a return value for the Requestor to keep track of it. This will be the only way for the requestor to validate responses he receives are well routed.

Reply:
I don't understand that. Can you provide an example of how the user of the library will use that?